pkgs/development/libraries/science/math/suitesparse/default.nix

```diff
@@ -47,48 +50,45 @@ stdenv.mkDerivation {
         -e 's|^[[:space:]]*\(NVCCFLAGS     =\)|NVCCFLAGS = $(NV20) -O3 -gencode=arch=compute_20,code=sm_20 -gencode=arch=compute_30,code=sm_30 -gencode=arch=compute_35,code=sm_35 -gencode=arch=compute_60,code=sm_60|'
   '';
 
-  makeFlags = [
-    "PREFIX=\"$(out)\""
-    "INSTALL_LIB=$(out)/lib"
-    "INSTALL_INCLUDE=$(out)/include"
-    "BLAS=-lopenblas"
-    "LAPACK="
-  ];
-
   NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin " -DNTIMER";
 
-  postInstall = ''
-    # Build and install shared library
+  buildPhase = ''
```

Comment:
Please invoke the pre- and post-phase hooks in a custom phase, i.e.
```.nix
{
buildPhase = ''
  runHook preBuild

  # ...

  runHook postBuild
'';
}
```
Other packages' setup hooks can add steps to the pre- and post-phase hooks which will not run otherwise. The same comment applies to `installPhase` below.

Reply:
Thanks, done.